### PR TITLE
various minor issues

### PR DESCRIPTION
**Pull request by rurban**

### CODE REVIEW

File: libdieharder/wyhash.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
+//Author: Wang Yi <godspeed_china@yeah.net>
```
Comment (marcusmueller) [inline libdieharder/wyhash.h:1]: Misses a license statement. Ability to distribute disputable!

**Review by marcusmueller — CHANGES_REQUESTED** [review]
Hi Reini,

I know this isn't my repo, but as maintainer of larger projects, I'd ask you to change a couple of things about this PR:

1. Please don't do a PR that contains fixes, cleanup, a version bump AND new functionality. It's impossible to review that. Could you split these into separate PRs?
2. you set `CFLAGS` in configure.ac. Not to used to autoconf, but I'd presume that's generally not a good way to achieve portability, as it stands to overwrite user settings
3. when you bump a release number, you do a release; it's kind of something rarely done in a PR, unless the one doing the PR is in charge of releases
4. I'm super excited about the CI inclusion, that should *so much* be a separate PR, as it really can't hurt anyone (that being said, the Travis CI commit removes the PS documentation, and that shouldn't be done "implicitly")
5. You checked in a lot of generated files, including output of validating your new RNGs. That feels wrong.
6. Well, many of your tests fail. I've got a xoroshiro128+ implementation that fails none but `diehard_sums` (which is a DONOTUSE test), so I don't think the state of your RNGs is satisfactory yet! That might just be a small issue (maybe in the float conversion? Maybe in the seeding?), but it's really hard to review that among all the other things.

File: dieharder.pc.in [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Name: dieharder
```
Comment (marcusmueller) [inline dieharder.pc.in:1]: :heart: awesome!

File: libdieharder/Makefile.am [hunk 3]
```diff
@@ -19,8 +19,8 @@ man3_MANS = libdieharder.3
 # Library definitions
 lib_LTLIBRARIES = libdieharder.la
 libdieharder_la_LIBADD = -lgsl -lgslcblas -lm
-libdieharder_la_LDFLAGS = -version-number @DIEHARDER_LT_VERSION@
-libdieharder_la_CFLAGS = -std=c99 -Wall -pedantic
+libdieharder_la_LDFLAGS = -no-undefined -version-number @DIEHARDER_LT_VERSION@
+libdieharder_la_CFLAGS = -std=gnu99 -Wall -pedantic
```
Comment (marcusmueller) [inline libdieharder/Makefile.am]: aaah, is that intentional? I thought the source was C99; I presume clang does gnu99 well enough, but changing the standard from something standard to something extended feels bad. What's the reasoning here? I think this is something you'd want to document.

File: dieharder/testbits.c [hunk 4]
```diff
@@ -32,16 +32,16 @@ void testbits()
...
- pattern[2] = 4;  /* grab 4 more */
- pattern[3] = -12;/* skip 12 */
- pattern[4] = 3;  /* grab 3 more */
- pattern[5] = 0;
+ //pattern[0] = 4;  /* grab 4 bits */
+ //pattern[1] = -8; /* skip 8 bits */
+ //pattern[2] = 4;  /* grab 4 more */
+ //pattern[3] = -12;/* skip 12 */
+ //pattern[4] = 3;  /* grab 3 more */
+ //pattern[5] = 0;
```
Comment (marcusmueller) [inline dieharder/testbits.c:44]: You're commenting out a full testpattern here. Why?

File: dieharder/parsecl.c [hunk 5]
```diff
@@ -125,8 +125,11 @@ void parsecl(int argc, char **argv)
         */
        dtest_tmp =  strtol(optarg,&endptr,10);
        /* printf("optarg = %s, dtest_tmp = %d endptr = %s\n",optarg,dtest_tmp,endptr); */
+       dtest_name[0] = '\0';
+       dtest_num = -1;
        if(strncmp(optarg,endptr,1) == 0){
-         strncpy(dtest_name,optarg,128);
+         strncpy(dtest_name,optarg,127);
```
Comment (marcusmueller) [inline dieharder/parsecl.c:134]: good catch. ASAN or manual?

File: dieharder/copyright.h [hunk 6]
```diff
@@ -12,6 +12,5 @@
  * DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
  * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  * PERFORMANCE OF THIS SOFTWARE.
- *
```
Comment (marcusmueller) [inline dieharder/copyright.h:15]: unless you're doing an overall code formatting commit, I'd avoid having such diminutive changes for the sake of change!

File: config.sub [hunk 7]
```diff
@@ -1,44 +1,40 @@
 #! /bin/sh
 # Configuration validation subroutine script.
-#   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
-#   2000, 2001, 2002, 2003, 2004, 2005, 2006 Free Software Foundation,
-#   Inc.
+#   Copyright 1992-2018 Free Software Foundation, Inc.
```
Comment (marcusmueller) [inline config.sub:3]: This is an auto-generated file. This should not be checked in to begin with!

File: autogen.sh [hunk 8]
```diff
@@ -10,7 +10,8 @@ case `uname` in
 esac
 $LIBTOOLIZE --copy
 
-autoheader
-aclocal
+#autoheader
+#aclocal
+autoreconf -fi
```
Comment (marcusmueller) [inline autogen.sh:15]: `autoreconf --install` does all the other steps, including automake and autoconf.

File: libdieharder/Makefile.am (hunk 3 above)
Comment (rurban) [reply libdieharder/Makefile.am]: Oh, I changed` -std=c99` to `-std=gnu99`? I was not aware of that. Probably it came from an external patch that I added.
Yes, it came with the eeaad20b8e690389c94d7b1a18f9d64c0eead909 cygwin patch by fd00.
I can try without.

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  - if [ -n "$DISTCHECK" ]; then
+      make -j4 distcheck && sha256sum dieharder-*.tar.* ;
+    fi
+
+deploy:
+  provider: releases
+  file_glob: true
+  file: dieharder-*.tar.*
+  api_key:
+    secure: "C86usbAdSxmRlFIwteeK+dyu7xASSG4y+jLpvipg/cw8SnRTkwWMIwmzXgbqA5uuskGs4jxyK0OKzrobg4KqnNUNa9wxJ/nLPFi9JxKDLwb9YxskoWigfYbdsXz/unU6XZ6giAsAbjiZQpgTl70DDlk9d1adQQGSkUb5CT6xmlmBuFF+OFubVNtMC3rhQoICS4O5a3uo5+qyY/FUjWJK39h5PfxUCW1VCrD2kuW8uoITKBmc/FAAxydFppwelKLMaI2B4O6pwpV0DBwTKhx6l/uQufDzwC8//hyDLmK7rcIy66YNVKUt/y+cwrpXgzlKsjvqNuHyeGE8N6alwBNU59g0QBJ1h+46aO8SSwRFHPLyLBkN2KY1VvaPTB2I0QYYoUAaACwXVcYpp1hb8RYRLNM6VCByeDhlTZNO9H2Nin2k1kqeGtnVYDbdKgPQL0cXpj4OKKDJlbNjv2hMVW0MPx92GaoXDhI3HfmmV66Qf4OyJT5rnX01kxeYzlkByUkG1i4FGB0xyg1OZSHd7B17KgPr0suVz0f9N2mFN1dGzp2O2alFwRl3GIpgCoAOlwsz5nJs55IqCOyaTvk7Zyh2PkeF6GKgFC2zgLL9g5D58+ZFulTojp5lshbOTepgvj3Oo+1rNoa6ke9F5Jvi0KOtVYM/z2wapr3pTa7bYcJK/lg="
```
Comment (marcusmueller) [inline .travis.yml:47]: did you actually mean to publish this?

File: libdieharder/wyhash.h (hunk 1 above)
Comment (rurban) [reply libdieharder/wyhash.h:1]: I added thre source in the commit: See https://github.com/wangyi-fudan/wyhash/blob/master/LICENSE

File: .travis.yml (hunk 9 above)
Comment (marcusmueller) [reply .travis.yml:47]: (I'm aware this is useless to anyone else for gaining access, but upstreaming a key that is only useful for you feels wrong)

**Comment by rurban** [comment]
I'll rather just close this PR and discuss it with upstream. I dont think Jochen wants to maintain it anyway.

Ad the xoroshiro fails: 
Could be the fast double-conversion. I used Vigna's recommended variant. Which is different to the dieharder conversion by divsion.
Also could be the seeding. I tried a good seed, better than Vigna's {1,-1, 0, 0} default seed. But I rather think it's just trash.

File: dieharder/parsecl.c (hunk 5 above)
Comment (rurban) [reply dieharder/parsecl.c:134]: manual

File: dieharder/testbits.c (hunk 4 above)
Comment (rurban) [reply dieharder/testbits.c:44]: because it's not used. see exit(0) before.

File: config.sub (hunk 7 above)
Comment (rurban) [reply config.sub:3]: True, but was added upstream. I just updated it

File: dieharder/copyright.h (hunk 6 above)
Comment (rurban) [reply dieharder/copyright.h:15]: Not for the sake of chamges.

**Comment by rurban** [comment]
> Ad 2) you set CFLAGS in configure.ac. Not to used to autoconf, but I'd presume that's generally not a good way to achieve portability, as it stands to overwrite user settings

Nope, I add to CFLAGS, and don't want to set AM_CFLAGS. Later I will probe for the best CFLAGS (via autoconf-archive macros), but for now I went with this mess. More important is to get it modern first. See https://github.com/rurban/dieharder/issues/1

**Pull request closed without merging**